acra/security-controls/zones.md

```diff
@@ -0,0 +1,146 @@
+```
+
+Now application should explicitly specify ZoneID in SQl query (works for MySQL and PostgreSQL): `SELECT "application_data"."id", 'DDDDDDDDjKjECtcRBDkmHVBh', "application_data"."data" FROM "application_data";`
+Here we placed ZoneID as string literal before encrypted column `data` and ZoneID value will be returned by database before encrypted data. So result rows will look similar as in example of [method 1](#method-1):
+```
+1 DDDDDDDDjKjECtcRBDkmHVBh \x2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103 
+```
+
+#### Method 3
+
+We can use SQL as API to communicate with AcraServer to decrypt data similar to AcraTranslator and construct custom result rows.  
+Query for PostgreSQL: 
+```
+SELECT 1, 
+       'DDDDDDDDjKjECtcRBDkmHVBh', 
+       '\x2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103'::BYTEA;
+```
+
+Query for MySQL: 
+```
+SELECT 1, 
+       'DDDDDDDDjKjECtcRBDkmHVBh', 
+       X'2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103';
+```
+Here we explicitly specify ZoneID as `DDDDDDDDjKjECtcRBDkmHVBh` and explicitly specify AcraStruct value as HEX literal. 
+So, database will return data as is and result rows will look similar to previous examples:
+
+
+## How Zones work
+
+When analysing the database output stream, AcraServer searches for certain strings called Zone Ids (also ZoneId and zoneid/zone_id in code). They let Acra know that within this record a private key corresponding to the Zone Id should be used for the actual decryption of AcraStructs. It will only work if the user explicitly formats the output for Zone Id to precede the AcraStruct.
```

Comment:
I believe this has been discussed in the preceding sections so this one can be removed.

Reply:
agree